File: backend/src/main/java/com/elearning/E_Learning/Platform/service/AuthService.java
package com.elearning.E_Learning.Platform.service;

import com.elearning.E_Learning.Platform.dto.AuthRequest;
import com.elearning.E_Learning.Platform.dto.AuthResponse;
import com.elearning.E_Learning.Platform.dto.RegisterRequest;

public interface AuthService {
    AuthResponse login(AuthRequest authRequest);
    AuthResponse register(RegisterRequest registerRequest);
} 

File: frontend/src/app/services/authService.ts
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/auth';

export interface RegisterRequest {
  name: string;
  username: string;
  email: string;
  password: string;
}

export interface LoginRequest {
  usernameOrEmail: string;
  password: string;
}

export interface AuthResponse {
  token: string;
  tokenType: string;
}

export const authService = {
  async register(data: RegisterRequest): Promise<AuthResponse> {
    const response = await axios.post(`${API_URL}/register`, data);
    if (response.data.token) {
      localStorage.setItem('token', response.data.token);
    }
    return response.data;
  },

  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await axios.post(`${API_URL}/login`, data);
    if (response.data.token) {
      localStorage.setItem('token', response.data.token);
    }
    return response.data;
  },

  logout(): void {
    localStorage.removeItem('token');
  },

  getCurrentToken(): string | null {
    return localStorage.getItem('token');
  }
};